### PR TITLE
Canvas: Button API Editor support setting parameters

### LINKED ISSUE #74220: Canvas: Button API Editor support setting parameters
Add ability to add key value parameters to API editor (including how parameters are sent to server (header vs query)
Screenshot 2023-08-31 at 11 48 30 AM

### PR DESCRIPTION
**Pull request by adela-almasan**
Adds parameters support to `APIEditor`.

https://github.com/grafana/grafana/assets/88068998/2f043647-42a1-4a41-861b-609de3a81bcd

Test Dashboard:
[buttonAPIDashboard.txt](https://github.com/grafana/grafana/files/12570158/buttonAPIDashboard.txt)

Fixes #74220

**Special notes for your reviewer:**
https://github.com/grafana/grafana/pull/74566 needs to be merged first

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Review by nmarrs — COMMENTED** [review]
I think we should consider updating the implementation to something we are comfortable with longer term - for this I think we should consider making how the param is sent in the API request a param by param setting vs global to make this feature less restrictive

File: public/app/plugins/panel/canvas/editor/element/QueryParamsEditor.tsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  onChange: (v: Array<[string, string]>) => void;
+  value: Array<[string, string]>;
+}
+
+export const QueryParamsEditor = ({ value, onChange }: Props) => {
+  const [paramName, setParamName] = useState('');
+  const [paramValue, setParamValue] = useState('');
+
+  const changeParamValue = ({ target }: FormEvent) => {
+    setParamValue((target as HTMLButtonElement).value);
```
[the remark this answers is not in the document]
Comment (nmarrs) [reply public/app/plugins/panel/canvas/editor/element/QueryParamsEditor.tsx]: Thanks @ashharrison90! Tested your suggestion out locally and that fixed both instances :) pushed up changes to PR so betterer is happy now 😄

File: public/app/features/canvas/elements/button.tsx [hunk 2]
```diff
@@ -24,6 +24,8 @@ export const defaultApiConfig: APIEditorConfig = {
   endpoint: '',
   method: HttpRequestMethod.POST,
   data: '{}',
+  paramsType: 'header',
+  params: [],
```
[the remark this answers is not in the document]
Comment (adela-almasan) [reply public/app/features/canvas/elements/button.tsx:28]: yes - imo this entire section needs a redesign (_postman_ hint, hint), but it's just what we've settled for this first version. As for the query/header parameters - currently you could send query parameters via endpoint and then define parameters in header, I don't think it's very restrictive, just unintuitive maybe.. 😬  But I've discussed this with @lukasztyrala and he suggested to have 2 separate sections - `Query parameters` and `Header parameters` and define/use as needed. I'll make this change and in the future maybe we could iterate and add improvements based on feedback.

**Comment by adela-almasan** [comment]
@nmarrs updated as per Luke's suggestion. I'll create a task for future improvements.
params

**Review by nmarrs — APPROVED** [review]
This approach seems reasonable, great job!

File: public/app/plugins/panel/canvas/editor/element/APIEditor.tsx [hunk 3]
```diff
@@ -183,6 +224,12 @@ export function APIEditor({ value, context, onChange }: Props) {
           <RadioButtonGroup value={value?.method} options={httpMethodOptions} onChange={onMethodChange} fullWidth />
         </InlineField>
       </InlineFieldRow>
+      <Field label="Query parameters">
+        <QueryParamsEditor value={value?.queryParams ?? []} onChange={onQueryParamsChange} />
+      </Field>
+      <Field label="Header parameters">
+        <QueryParamsEditor value={value?.headerParams ?? []} onChange={onHeaderParamsChange} />
```
Comment (nmarrs) [inline public/app/plugins/panel/canvas/editor/element/APIEditor.tsx:231]: Nit: maybe we should update naming of this to be just "ParamsEditor"?

**Pull request merged**